src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -647,7 +647,7 @@ private static int FindMismatch(ReadOnlySpan<byte> span, ReadOnlySpan<byte> lite
             return indexOfFirstMismatch;
         }
 
-        private JsonException GetInvalidLiteralMultiSegment(ReadOnlySpan<byte> span)
+        private JsonException GetInvalidLiteralMultiSegment(scoped Span<byte> span)
```

Comment:
Why is this being changed to Span from ReadOnlySpan?

Reply:
It reduces code size at the callsite and also seems to work around a JIT bug in relation to cold code regions and inlining - I haven't gotten to getting a minimal repro for that yet, but will create a separate issue for that once I get it.

Reply:
Created this for the JIT codegen issue: https://github.com/dotnet/runtime/issues/115918

Reply:
Sorry, revert what?

Reply:
Revert the change being made so that it continues using ReadOnlySpan